### PR TITLE
Fixed the enum serialization issue

### PR DESCRIPTION
**Pull request by ghostdevhv**
Reference: https://github.com/Azure/pipelines/issues/41

### CODE REVIEW

**Comment by satyaye** [comment]
@kaverma - Can you or someone from your team please review and approve these changes? Looking for some quick help

**Comment by satyaye** [comment]
@ds-ms - Tagging you here as suggested by @kaverma in slack DM, can you pleas review this PR?

**Comment by ghostdevhv** [comment]
@ds-ms - Gentle reminder! Can you please review this PR? 
cc: @satyaye

File: lib/pipeline.runner.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ class PipelineRunner {
                 },
                 sourceBranch: sourceBranch,
                 sourceVersion: sourceVersion,
-                reason: BuildInterfaces.BuildReason.Triggered,
+                reason: buildReason,
```
Comment (ds-ms) [inline lib/pipeline.runner.js:103]: Update the TS files as well

Comment (ghostdevhv) [reply lib/pipeline.runner.js:103]: Done.

File: src/pipeline.runner.ts [hunk 2]
```diff
@@ -91,7 +92,7 @@ export class PipelineRunner {
             },
             sourceBranch: sourceBranch,
             sourceVersion: sourceVersion,
-            reason: BuildInterfaces.BuildReason.Triggered,
+            reason: buildReason,
```
Comment (ds-ms) [inline src/pipeline.runner.ts:95]: Shouldn't you add this state to the Enum?

**Comment by ds-ms** [comment]
@ghostdevhv could you rebase your branch, that will run the tests again

**Review by ds-ms — APPROVED** [review]

**Pull request merged**